aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -536,10 +536,12 @@ module aptos_token::token {
                     &mut collections.collection_data,
                     token_id.token_data_id.collection
                 );
-                collection_data.supply = collection_data.supply - 1;
-                // delete the collection data if the collection supply equals 0
-                if (collection_data.supply == 0) {
-                    destroy_collection_data(table::remove(&mut collections.collection_data, collection_data.name));
+                if (collection_data.maximum > 0) {
+                    collection_data.supply = collection_data.supply - 1;
+                    // delete the collection data if the collection supply equals 0
```

Comment:
Does it make sense to delete the collection? This would force the creator to have to re-create it later if they want to mint the tokens, which costs a lot more gas.

Reply:
create collection only create a collectData, which is about 0.002 APT as well. we probably want to be more proactive in cleaning unused data now since there is no rent

Reply:
That's a good point.